pkg/bgpv1/manager/policy_reconciler.go

```diff
@@ -0,0 +1,236 @@
+
+	// reconciliation successful, update the cache of configured policies
+	params.CurrentServer.Policies = desiredPolicies
+	return nil
+}
+
+func (r *PolicyReconciler) routeAttributesToPolicy(attrs v2alpha1api.CiliumBGPRouteAttributes, neighborAddress string, params ReconcileParams) (*types.Policy, error) {
+	var v4Prefixes, v6Prefixes types.PolicyPrefixMatchList
+
+	policy := &types.Policy{
+		Name: routeAttributesPolicyName(attrs, neighborAddress),
+		Type: types.PolicyTypeExport,
+	}
+
+	labelSelector, err := slim_metav1.LabelSelectorAsSelector(attrs.Selector)
+	if err != nil {
+		return nil, fmt.Errorf("failed constructing LabelSelector: %w", err)
+	}
+
+	switch attrs.SelectorType {
+	case v2alpha1api.PoolKindDefinition:
+		for _, pool := range r.ipPoolStore.List() {
+			if pool.Spec.Disabled {
+				continue
+			}
+			if attrs.Selector != nil && !labelSelector.Matches(labels.Set(pool.Labels)) {
+				continue
+			}
+			for _, cidrBlock := range pool.Spec.Cidrs {
+				cidr, err := netip.ParsePrefix(string(cidrBlock.Cidr))
+				if err != nil {
+					return nil, fmt.Errorf("failed to parse IPAM pool CIDR %s: %w", cidrBlock.Cidr, err)
+				}
+				if cidr.Addr().Is4() {
+					v4Prefixes = append(v4Prefixes, &types.PolicyPrefixMatch{CIDR: cidr, PrefixLenMin: maxPrefixLenIPv4, PrefixLenMax: maxPrefixLenIPv4})
+				} else {
+					v6Prefixes = append(v6Prefixes, &types.PolicyPrefixMatch{CIDR: cidr, PrefixLenMin: maxPrefixLenIPv6, PrefixLenMax: maxPrefixLenIPv6})
+				}
+			}
+		}
+	case v2api.CNKindDefinition:
+		if attrs.Selector != nil && !labelSelector.Matches(labels.Set(params.Node.Labels)) {
+			break
+		}
+		for _, podCIDR := range params.Node.ToCiliumNode().Spec.IPAM.PodCIDRs {
+			cidr, err := netip.ParsePrefix(podCIDR)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse PodCIDR %s: %w", podCIDR, err)
+			}
+			if cidr.Addr().Is4() {
+				v4Prefixes = append(v4Prefixes, &types.PolicyPrefixMatch{CIDR: cidr, PrefixLenMin: cidr.Bits(), PrefixLenMax: cidr.Bits()})
+			} else {
+				v6Prefixes = append(v6Prefixes, &types.PolicyPrefixMatch{CIDR: cidr, PrefixLenMin: cidr.Bits(), PrefixLenMax: cidr.Bits()})
+			}
+		}
+	default:
+		return nil, fmt.Errorf("invalid policy SelectorType: %s", attrs.SelectorType)
+	}
+
+	// sort prefixes to have consistent order for DeepEqual
+	sort.Slice(v4Prefixes, v4Prefixes.Less)
+	sort.Slice(v6Prefixes, v6Prefixes.Less)
+
+	// sort communities to have consistent order for DeepEqual
+	var communities []string
+	for _, c := range attrs.Communities {
+		communities = append(communities, string(c))
+	}
+	sort.Strings(communities)
+
+	if len(v4Prefixes) > 0 {
+		policy.Statements = append(policy.Statements, policyStatement(neighborAddress, v4Prefixes, attrs.LocalPreference, communities))
+	}
+	if len(v6Prefixes) > 0 {
+		policy.Statements = append(policy.Statements, policyStatement(neighborAddress, v6Prefixes, attrs.LocalPreference, communities))
+	}
```

Comment:
If we merge all rules into single Statements list, do we need to differentiate IPv4 and IPv6?

Reply:
Yeah we need to differentiate here - families can not be mixed in a single Statement, but it is possible to have Statements with different families in a single policy. Will add a comment clarifying this.

Reply:
Left a comment about this in the code.